Source/WebCore/platform/network/cocoa/CookieCocoa.mm

```diff
@@ -36,17 +36,16 @@
 static Vector<uint16_t> portVectorFromList(NSArray<NSNumber *> *portList)
 {
     return Vector<uint16_t>(portList.count, [portList](size_t i) {
-        NSNumber *port = portList[i];
-        return port.unsignedShortValue;
+        return portList[i].unsignedShortValue;
```

Comment:
Changes like this make me wonder about the static analyzer’s quality. The new code is exactly as safe as the old code, but somehow does not yield the same warning. If we need a `RetainPtr`, whether it’s an intermediate value in an expression or a local variable doesn’t matter.

Reply:
cc @rniwa to comment. I do think that it shouldn't have required a `RetainPtr`.

Reply:
Hm... I'm a bit surprised that the static analyzer doesn't generate a warning after this change. I would have expected it to generate a call argument warning since any Obj-C property access like `unsignedShortValue` is a selector invocation.

Reply:
ok, I think we'd get a call argument warning after this change like this:
```
CookieCocoa.mm:39:16: warning: Reciever is unretained and unsafe [alpha.webkit.UnretainedCallArgsChecker]
   39 |         return portList[i].unsignedShortValue;
      |                ^~~~~~~~~~~
```

This is expected because `unsignedShortValue` is a Objective-C message send and can have side effects.

Reply:
Maybe it will generate a warning. I haven't un-skipped files yet because I was waiting for Brianna to fix the bot.

Reply:
So there are a couple of options here. We could deploy `RetainPtr` here out of principle. Alternatively, we could hard code a list of properties for `NSNumber` and other basic NS classes and allow them to be invoked without a `RetainPtr`.

Reply:
Yet another option is to manually suppress this warning for this line because we know at least the common implementation of `unsignedShortValue` doesn't have a side effect.

It's a bit of risky business though because ObjC selectors can be overridden / fiddled with for a specific object. It's possible that someone could create a new subclass of `NSNumber` which implements `unsignedShortValue` with a serious side effect.

Reply:
I will retain.